src/apps/material/helper.ts

```diff
@@ -402,3 +410,75 @@ export const isParallelReservation = (manifestations: Manifestation[]) =>
   manifestations.length > 1 &&
   hasCorrectAccessType(AccessTypeCode.Physical, manifestations) &&
   !isArticle(manifestations);
+
+// Because we  need to exclude the branches that are blacklisted, we need to use a custom hook to prevent duplicate code
+const getBlacklistedArgs = (
+  faustIds: FaustId[],
+  config: UseConfigFunction,
+  blacklist:
+    | "blacklistedAvailabilityBranchesConfig"
+    | "blacklistedPickupBranchesConfig"
+) => {
+  const blacklistBranches = config(blacklist, {
+    transformer: "stringToArray"
+  });
+  return {
+    recordid: faustIds,
+    ...(blacklistBranches ? { exclude: blacklistBranches } : {})
+  };
+};
+
+export const useGetAvailability = ({
```

Comment:
Nice initiative. You could also consider moving the functionality around availability to a separate file in core somewhere. core/utils maybe?
Since it is more global than just the material app in my opinion.